Classes/Plugin/Eid/SearchSuggest.php

```diff
@@ -39,25 +38,34 @@ class SearchSuggest
      */
     public function main(ServerRequestInterface $request)
     {
+        $output = [];
+        // Get input parameters and decrypt core name.
         $parameters = $request->getParsedBody();
-        $encrypted = (string)$parameters['encrypted'];
-        $hashed = (string)$parameters['hashed'];
+        $encrypted = (string) $parameters['encrypted'];
+        $hashed = (string) $parameters['hashed'];
         if (empty($encrypted) || empty($hashed)) {
             throw new \InvalidArgumentException('No valid parameter passed!', 1580585079);
         }
         $core = Helper::decrypt($encrypted, $hashed);
-
-        $output = '';
-        if (!empty($core)) {
-            $query = (string)$parameters['q'];
-            $url = trim(Solr::getSolrUrl($core), '/') . '/suggest/?wt=xml&q=' . Solr::escapeQuery($query);
-            $output = GeneralUtility::getUrl($url);
+        // Perform Solr query.
+        $solr = Solr::getInstance($core);
+        if ($solr->ready) {
+            $query = $solr->service->createSelect();
+            $query->setHandler('suggest');
+            $query->setQuery(Solr::escapeQuery((string) $parameters['q']));
+            $query->setRows(0);
+            $results = $solr->service->select($query)->getResponse()->getBody();
+            $result = json_decode($results);
+            foreach ($result->spellcheck->suggestions as $suggestions) {
+                foreach ($suggestions->suggestion as $suggestion) {
```

Comment:
This does through an exception because the first array element is no object.

That's because of the result of json_decode().
Response from Solr:
`/suggest?q=dresde&json.nl=flat&omitHeader=true&fl=*,score&start=0&rows=0&wt=json`
```
{
  "spellcheck":{
    "suggestions":[
      "dresde",{
        "numFound":2,
        "startOffset":0,
        "endOffset":6,
        "suggestion":["dresdener pflasterstein-werke aktiengesellschaft, dresden-leubnitz",
          "dresden-leipziger schnellpressenfabrik ag, coswig"]}]}}
```
Result of json_decode():
```
object(stdClass)#3 (1) {
  ["spellcheck"]=>
  object(stdClass)#2 (1) {
    ["suggestions"]=>
    array(2) {
      [0]=>
      string(6) "dresde"
      [1]=>
      object(stdClass)#1 (4) {
        ["numFound"]=>
        int(2)
        ["startOffset"]=>
        int(0)
        ["endOffset"]=>
        int(6)
        ["suggestion"]=>
        array(2) {
          [0]=>
          string(66) "dresdener pflasterstein-werke aktiengesellschaft, dresden-leubnitz"
          [1]=>
          string(49) "dresden-leipziger schnellpressenfabrik ag, coswig"
        }
      }
    }
  }
}

```
If we use json_decode with the optional paramter to get an associative array, the result would be:
```
array(1) {
  ["spellcheck"]=>
  array(1) {
    ["suggestions"]=>
    array(2) {
      [0]=>
      string(6) "dresde"
      [1]=>
      array(4) {
        ["numFound"]=>
        int(2)
        ["startOffset"]=>
        int(0)
        ["endOffset"]=>
        int(6)
        ["suggestion"]=>
        array(2) {
          [0]=>
          string(66) "dresdener pflasterstein-werke aktiengesellschaft, dresden-leubnitz"
          [1]=>
          string(49) "dresden-leipziger schnellpressenfabrik ag, coswig"
        }
      }
    }
  }
}

```
So we must check for a valid object or array before calling the foreach() loop.